provider/azure/environ.go

```diff
@@ -1232,6 +1241,7 @@ func (env *azureEnviron) deleteVirtualMachine(
 	logger.Debugf("- deleting virtual machine (%s)", vmName)
 	vmResultCh, errCh := vmClient.Delete(env.resourceGroup, vmName, nil)
 	if result, err := <-vmResultCh, <-errCh; err != nil {
+		errorutils.HandleCredentialError(err, ctx)
```

Comment:
If there's a credential error here, any subsequent call to GetContainerReference below will also fail right? Did we need to do something like:
if maybeStorageClient && validCredential {

of something